plugins/inputs/http_listener_v2/http_listener_v2.go

```diff
@@ -268,6 +269,17 @@ func (h *HTTPListenerV2) collectBody(res http.ResponseWriter, req *http.Request)
 		return nil, false
 	}
 
+	// Handle snappy request bodies
+	if req.Header.Get("Content-Encoding") == "snappy" {
```

Comment:
About 20 lines above there's a similar block for gzip.  It looks like we should change the "if content-encoding == gzip" to a switch statement that also handles snappy, then move this new block into the switch statement.  The snappy decoding needs to happen at the same place gzip does, namely before the MaxBytesReader.

Reply:
The NewReader function of snappy would return a reader such as the gzip implementation so the switch would work but the block implementation of snappy that we need (for prometheus remote write) is a Decode which takes in bytes and returns bytes. So I think it needs to be after the reader on 266 unless I'm missing something.